rest-api/src/main/java/io/quarkus/qe/exceptions/CatalogException.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.qe.exceptions;
+
+@FunctionalInterface
+public interface CatalogException {
+
+    default CatalogError toCatalogError(String msg) {
+        return CatalogError.builder().withCode(getErrorCode()).withMsg(msg).build();
+    }
+
+    default CatalogError toCatalogError() {
+        return CatalogError.builder().withCode(getErrorCode()).build();
+    }
```

Comment:
The "toCatalogError"s methods are builders that should not be part of this interface. (I think the proper place is to move them to the CatalogExceptionMapper.java), what do you think?

Also, I think it makes sense that this interface is a custom exception instead:

```java
public class CatalogException extends Exception {

    private final int errorCode;

    public CatalogException(int errorCode) {
        this.errorCode = errorCode;
    }

    public int getErrorCode() {
        return errorCode;
    }
   
}
```

This way, the contract is clear and we don't need to have things like this:

```java
public class RepositoryAlreadyExistsException extends Exception implements CatalogException {
```

Just:

```java
public class RepositoryAlreadyExistsException extends CatalogException {
```